### PR TITLE
Completed documentation for many visual nodes

### PR DESCRIPTION
**Pull request by clayjohn**
Finish class reference documentation for:
- ArrayMesh
- Geometry
- GeometryInstance
- Image
- Light
- Mesh
- MeshInstance
- MeshInstance2D
- MultiMeshInstance
- MultiMeshInstance2D
- OmniLight
- Texture3D
- VisualInstance

I plan on updating all the visual type stuff over the next couple days.

### CODE REVIEW

File: doc/classes/Mesh.xml [hunk 1]
```diff
@@ -128,52 +129,76 @@
...
+			Flag used to mark a compressed (half float) weight array.
 		</constant>
 		<constant name="ARRAY_COMPRESS_INDEX" value="131072" enum="ArrayFormat">
+			Flag used to mark a compressed index array.
 		</constant>
 		<constant name="ARRAY_FLAG_USE_2D_VERTICES" value="262144" enum="ArrayFormat">
+			Flag used to mark that the array contains 2D vertices.
 		</constant>
 		<constant name="ARRAY_FLAG_USE_16_BIT_BONES" value="524288" enum="ArrayFormat">
+			Flag used to mark that the array contains 2D vertices.
```
Comment (akien-mga) [inline doc/classes/Mesh.xml]: Copy paste mistake, that's the same as previous.

File: doc/classes/Light.xml [hunk 2]
```diff
@@ -67,38 +69,53 @@
...
 		<constant name="PARAM_SHADOW_SPLIT_1_OFFSET" value="9" enum="Param">
+			Constant for accessing [member DirectionalLight.shadow_split_1].
 		</constant>
 		<constant name="PARAM_SHADOW_SPLIT_2_OFFSET" value="10" enum="Param">
+			Constant for accessing [member DirectionalLight.shadow_split_2].
 		</constant>
 		<constant name="PARAM_SHADOW_SPLIT_3_OFFSET" value="11" enum="Param">
+			Constant for accessing [member DirectionalLight.shadow_split_3].
 		</constant>
 		<constant name="PARAM_SHADOW_NORMAL_BIAS" value="12" enum="Param">
```
Comment (akien-mga) [inline doc/classes/Light.xml]: You missed one here :)

File: doc/classes/ArrayMesh.xml [hunk 3]
```diff
@@ -185,6 +186,7 @@
 	</methods>
 	<members>
 		<member name="blend_shape_mode" type="int" setter="set_blend_shape_mode" getter="get_blend_shape_mode" enum="Mesh.BlendShapeMode" default="1">
+			Sets the blend shape mode to one of [enum Mesh.BlendShapeMode]. See [enum Mesh.BlendShapeMode] for options.
```
Comment (akien-mga) [inline doc/classes/ArrayMesh.xml]: Maybe a bit redundant to mention the enum twice? Doesn't hurt though, and I guess you did it to be consistent with the "See enum for options." format.

File: doc/classes/OmniLight.xml [hunk 4]
```diff
@@ -33,8 +33,10 @@
 			Shadows are rendered to a cubemap. Slower than [constant SHADOW_DUAL_PARABOLOID], but higher-quality.
 		</constant>
 		<constant name="SHADOW_DETAIL_VERTICAL" value="0" enum="ShadowDetail">
+			Use more detail vertically when computing the shadow.
 		</constant>
 		<constant name="SHADOW_DETAIL_HORIZONTAL" value="1" enum="ShadowDetail">
+			Use more detail horizontally when computing computing the shadow.
```
Comment (akien-mga) [inline doc/classes/OmniLight.xml]: "computing" twice

File: doc/classes/GeometryInstance.xml [hunk 5]
```diff
@@ -15,6 +15,7 @@
 			<argument index="0" name="flag" type="int" enum="GeometryInstance.Flags">
 			</argument>
 			<description>
+				Returns the [enum GeometryInstance.Flags] that have been set for this object. See [enum GeometryInstance.Flags] for options.
```
Comment (akien-mga) [inline doc/classes/GeometryInstance.xml]: Same as above - but similarly, it's also fine to keep as is.

File: doc/classes/Image.xml [hunk 6]
```diff
@@ -502,8 +504,10 @@
 			The maximal height allowed for [Image] resources.
 		</constant>
 		<constant name="FORMAT_L8" value="0" enum="Format">
+			Texture format with a single 8 bit depth representing luminance.
```
Comment (akien-mga) [inline doc/classes/Image.xml]: "8-bit" with a hyphen as adjective.

File: doc/classes/Light.xml [hunk 7]
```diff
@@ -67,38 +69,53 @@
 			If [code]true[/code], the light will cast shadows.
 		</member>
 		<member name="shadow_reverse_cull_face" type="bool" setter="set_shadow_reverse_cull_face" getter="get_shadow_reverse_cull_face" default="false">
+			If [code]true[/code], reverses the backface culling of the mesh. This can be useful when you have a flat mesh that has a light behind it. If you need to cast a shadow on both sides of the mesh, set the mesh to use double sided shadows with [constant GeometryInstance.SHADOW_CASTING_SETTING_DOUBLE_SIDED].
```
Comment (akien-mga) [inline doc/classes/Light.xml]: "double-sided" with hyphen as adjective.

File: doc/classes/VisualInstance.xml [hunk 8]
```diff
@@ -18,12 +20,14 @@
...
 			</return>
 			<description>
+				Returns the RID of the resource associated with this VisualInstance. For example, if the Node is a [MeshInstance], this will return the RID of the associated [Mesh]. Equivalent to [method VisualServer.instance_get_base].
 			</description>
 		</method>
 		<method name="get_instance" qualifiers="const">
 			<return type="RID">
 			</return>
 			<description>
+				Returns the RID of this instance. This RID is the same as the RID returned by [method VisualServer.instance_create()]. This RID is needed if you want to call [VisualServer] functions directly on this VisualInstance.
```
Comment (akien-mga) [inline doc/classes/VisualInstance.xml]: There shouldn't be `()` in the `[method ...]` markup.

File: doc/classes/VisualInstance.xml [hunk 9]
```diff
@@ -32,6 +36,7 @@
 			<argument index="0" name="layer" type="int">
 			</argument>
 			<description>
+				Returns [code]true[/code] when specified layer is enabled in [member layers] and [code]false[/code] otherwise.
```
Comment (akien-mga) [inline doc/classes/VisualInstance.xml]: "when **the** specified layer" I think?

File: doc/classes/VisualInstance.xml [hunk 10]
```diff
@@ -48,8 +53,7 @@
 			<argument index="0" name="base" type="RID">
 			</argument>
 			<description>
-				Sets the base of the VisualInstance, which changes how the engine handles the VisualInstance under the hood.
-				It is recommended to only use [method set_base] if you know what you're doing.
+				Sets the resource that is instanced by this VisualInstance, which changes how the engine handles the VisualInstance under the hood. Equivalent to [method VisualServer.instance_set_base].
```
Comment (akien-mga) [inline doc/classes/VisualInstance.xml]: I think it should be "instantiated", but this wrong verb form is pretty common in our API :)

**Review by akien-mga — APPROVED** [review]
A few nitpicks, but otherwise great work!

**Comment by akien-mga** [comment]
A few errors:
```
ERROR: Unresolved member 'OmniLight.omni_attentuation', file: Light
ERROR: Unresolved member 'DirectionalLight.shadow_max_distance', file: Light
ERROR: Unresolved member 'DirectionalLight.shadow_split_1', file: Light
ERROR: Unresolved member 'DirectionalLight.shadow_split_2', file: Light
ERROR: Unresolved member 'DirectionalLight.shadow_split_3', file: Light
ERROR: Unresolved member 'DirectionalLight.shadow_bias_split_scale', file: Light
ERROR: Unresolved method 'VisualServer.instance_get_base', file: VisualInstance
ERROR: Unresolved method 'VisualServer.instance_create()', file: VisualInstance
```

File: doc/classes/ArrayMesh.xml (hunk 3 above)
Comment (clayjohn) [reply doc/classes/ArrayMesh.xml]: It is a bit awkward. I'll remove it.

**Comment by clayjohn** [comment]
Thanks! I blame all the random mistakes on the fact that I did this while in class today... oops!

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**